umodbus/serial.py

```diff
@@ -231,63 +263,144 @@ def _uart_read_frame(self, timeout: Optional[int] = None) -> bytearray:
+        )
+        return self._send(modbus_pdu=modbus_pdu, slave_addr=slave_addr)
 
-        if self._ctrlPin:
-            total_frame_time_us = self._t1char * len(serial_pdu)
-            while time.ticks_us() <= send_start_time + total_frame_time_us:
-                machine.idle()
-            self._ctrlPin(0)
+    def send_exception_response(self,
+                                slave_addr: int,
+                                function_code: int,
+                                exception_code: int) -> Optional[Awaitable]:
+        """
+        Send an exception response to a client.
 
-    def _send_receive(self,
-                      modbus_pdu: bytes,
-                      slave_addr: int,
-                      count: bool) -> bytes:
+        :param      slave_addr:      The slave address
+        :type       slave_addr:      int
+        :param      function_code:   The function code
+        :type       function_code:   int
+        :param      exception_code:  The exception code
+        :type       exception_code:  int
+
+        :returns:   Request response - None for a synchronous server, or
+                    an awaitable for an asynchronous server due to AsyncRequest
+        :rtype      Awaitable, optional
         """
-        Send a modbus message and receive the reponse.
+        modbus_pdu = functions.exception_response(
+            function_code=function_code,
+            exception_code=exception_code)
+        return self._send(modbus_pdu=modbus_pdu, slave_addr=slave_addr)
 
-        :param      modbus_pdu:  The modbus Protocol Data Unit
-        :type       modbus_pdu:  bytes
-        :param      slave_addr:  The slave address
-        :type       slave_addr:  int
-        :param      count:       The count
-        :type       count:       bool
 
-        :returns:   Validated response content
-        :rtype:     bytes
+class Serial(CommonRTUFunctions, CommonModbusFunctions):
+    """Modbus Serial/RTU client"""
+
+    def _exit_read(self, response: bytearray) -> bool:
         """
-        # flush the Rx FIFO
-        self._uart.read()
+        Return on modbus read error
 
-        self._send(modbus_pdu=modbus_pdu, slave_addr=slave_addr)
+        :param      response:    The response
+        :type       response:    bytearray
 
-        return self._validate_resp_hdr(response=self._uart_read(),
-                                       slave_addr=slave_addr,
-                                       function_code=modbus_pdu[0],
-                                       count=count)
+        :returns:   State of basic read response evaluation
+        :rtype:     bool
+        """
+        if response[1] >= Const.ERROR_BIAS:
+            if len(response) < Const.ERROR_RESP_LEN:
+                return False
+        elif (Const.READ_COILS <= response[1] <= Const.READ_INPUT_REGISTER):
+            expected_len = Const.RESPONSE_HDR_LENGTH + 1 + \
+                           response[2] + Const.CRC_LENGTH
```

Comment:
continuation line over-indented for hanging indent

Ususally the test pipeline should find this flake8 issues, not sure why it does not run at all

Reply:
Okay, fair enough. It's only activated for branches. Not for `pull_requests`

Reply:
I'm not sure what the correct way to fix it is since flake8 doesn't signal any error for me for some reason. I've indented line 373 1 more than line 372, but let me know if I made a mistake.